docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+Hypothesis CSS Style Guide
+==========================
+
+Hypothesis' CSS generally follows the principles of [Trello's CSS Style Guide](https://github.com/trello/trellisheets)
+for writing maintainable styling. In particular:
+
+ * Use simple, non-nested class selectors where possible
+ * Organize CSS into components. Unlike Trello, we use BEM-style naming (see below) and do not indent selectors
+ * Be careful with CSS preprocessor features. Unlike Trello, we use SASS but the principle
+   of using 'vanilla' CSS and being careful with preprocessor features still applies.
+   See the note about `@extends` below
+ * Use naming conventions to indicate which classes are purely presentational and which are referenced in code
+
+## File Structure
+
+The SASS for Hypothesis projects is organized into several parts:
+
+ * A 'root' file which imports all of the components used in the site, structured like this:
```

Comment:
> Doesn't this mean that all of our CSS is downloaded for every page?

As Nick says, there can be more than one root file. eg. We don't serve the CSS for our admin pages to ordinary visitors to the site. This _does_ still mean that typically you would download more CSS than you need for any given page load, but it would save fetching another bundle when navigating to a different page.

This is a classic trade-off with bundling assets together. With HTTP 2 one can generally afford to use smaller, more granular bundles because the overhead of an individual request is lower and there isn't the same limit on the number of concurrent requests to a particular domain.